lib/src/source_visitor.dart

```diff
@@ -108,12 +108,45 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   visitAdjacentStrings(AdjacentStrings node) {
     builder.startSpan();
-    builder.startRule();
-    visitNodes(node.strings, between: splitOrNewline);
+    var rule = new Rule();
+    builder.startRule(rule);
+
+    var chunk = zeroSplit();
+    rule.imply(chunk.rule);
+
+    visit(node.strings.first);
+    var previous = node.strings.first;
+    for (var string in node.strings.skip(1)) {
+      // Always split after a string that ends in a newline. Otherwise, indent a
+      // little bit so it's clear that the next string is a continuation rather
+      // than a new line.
+      if (_endsWithNewline(previous)) {
+        builder.writeWhitespace(Whitespace.nestedNewline);
+      } else {
+        builder.indent();
+        split();
+        builder.unindent();
+      }
```

Comment:
> I think this policy is a little too specific for dartfmt. I agree it looks nice in the pub examples, but it's meaningless in lots of other examples where the strings in question are not lines of human readable text. If you look at the changed regression tests, there are strings of JSON and other stuff like that where the extra indentation is just confusing.

I strongly disagree. Based on the SDK, the vast majority of adjacent string literals are text where the distinction between a newline and a continued line is relevant to the reader. Even in cases where it's not relevant to the format itself, having this as a universal convention helps communicate to a reader that the string being constructed will exist all on one line.

Also, the current lack of a policy isn't any more meaningful. The regression test for #21 [wipes out the prior formatting entirely](https://github.com/dart-lang/dart_style/blob/master/test/regression/0000/0021.stmt#L13-L49). Having some policy that works well in the substantial majority of cases is better than having no policy at all—isn't that the entire point of the formatter?

> Also, this is what's causing the above asserts to fire. Indentation is for statement-level stuff and precedes expression nesting. Once you're inside an expression, messing with the indentation is likely to not do what you want.

I'm not sure I follow. What I want here is a little extra indentation for this particular split, which happens to be inside an expression. What else should I use to get that effect?